extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/RestClientMetricsFilter.java

```diff
@@ -0,0 +1,104 @@
+package io.quarkus.micrometer.runtime.binder;
+
+import javax.inject.Inject;
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientRequestFilter;
+import javax.ws.rs.client.ClientResponseContext;
+import javax.ws.rs.client.ClientResponseFilter;
+import javax.ws.rs.ext.Provider;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.Unremovable;
+
+/**
+ * A client filter for the JAX-RS Client and MicroProfile REST Client that records OpenTelemetry data.
+ */
+@Unremovable
+@Provider
```

Comment:
This will get wired only at runtime when JAX-RS starts... Just thinking out loud.